libraries/core/src/descriptor/validate.rs

```diff
@@ -6,28 +6,44 @@ use crate::{
 };
 
 use eyre::{bail, eyre, Context};
+use std::collections::HashSet;
 use std::{path::Path, process::Command};
 use tracing::info;
 
-use super::{resolve_path, Descriptor, SHELL_SOURCE};
+use super::{resolve_path, source_to_path, Descriptor, SHELL_SOURCE};
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
 pub fn check_dataflow(dataflow: &Descriptor, working_dir: &Path) -> eyre::Result<()> {
     let nodes = dataflow.resolve_aliases_and_set_defaults()?;
     let mut has_python_operator = false;
+    let is_multiple = nodes
```

Comment:
Could we rename this, `is_local` and check `if is_local`?

It might be easier to read and less confusing.

Sorry to be picky :sweat_smile: